android/src/main/kotlin/io/material/plugins/dynamic_color/DynamicColorPlugin.kt

```diff
@@ -38,15 +38,40 @@ class DynamicColorPlugin : FlutterPlugin, MethodCallHandler {
       } else {
         result.success(null)
       }
+    } else if (call.method.equals("getAccentColor")) {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        val resources: Resources = binding.applicationContext.resources
+        val theme: Resources.Theme = binding.applicationContext.theme
+        result.success(getAccentColor(resources, theme))
+      } else {
+        result.success(null)
+      }
     } else {
       result.notImplemented()
     }
   }
 
-  override fun onDetachedFromEngine(@NonNull binding: FlutterPluginBinding) {
+  override fun onDetachedFromEngine(binding: FlutterPluginBinding) {
     channel.setMethodCallHandler(null)
   }
 
+  @RequiresApi(Build.VERSION_CODES.Q)
+  private fun getAccentColor(resources: Resources, theme: Resources.Theme): Int {
+    val resId = when (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) {
+      Configuration.UI_MODE_NIGHT_YES -> android.R.style.Theme_DeviceDefault
+      else -> android.R.style.Theme_DeviceDefault_Light
+    }
+
+    val array : TypedArray = theme.obtainStyledAttributes(
+      resId,
+      IntArray(1){android.R.attr.colorAccent},
+    )
+
+    @ColorInt val color = array.getColor(0, 0)
```

Comment:
do you want black as default accent color for both light and dark?

Reply:
If you do `context.getTheme().obtainStyledAttributes(new int[] {android.R.attr.colorAccent})`, you shouldn't have to worry about which default theme vs custom theme at all, since the theme is based on the context. But if you're having issues retrieving the theme from context, this is fine for now.